contracts/PToken.sol

```diff
@@ -9,71 +9,139 @@ contract PToken is ERC20UpgradeSafe, OwnableUpgradeSafe {
   using SafeMath for uint256;
   using SafeERC20 for IERC20;
 
-  IERC20 public acceptedToken; // ERC20 that is used for purchasing pToken
-  uint256 public price; // The amount of aToken to purchase pToken
+  /// @notice ERC20 that is used for purchasing pTokens
+  IERC20 public acceptedToken;
 
-  event Initialized(address owner, uint256 price, uint256 supply);
+  /// @notice Price per pToken, denominated in units of `acceptedToken`
+  uint256 public price;
 
-  event Purchased(address buyer, uint256 cost, uint256 amountReceived);
+  /// @dev Emitted when pToken is initialized
+  event Initialized(uint256 price, uint256 supply);
 
-  event Redeemed(address seller, uint256 amountRedeemed);
+  /// @dev Emitted when pTokens are purchased
+  event Purchased(address buyer, uint256 cost, uint256 amount);
 
-  event PriceUpdated(address owner, uint256 newPrice);
+  /// @dev Emitted when pTokens are redeemed
+  event Redeemed(address seller, uint256 amount);
 
-  event Minted(address owner, uint256 amountMinted);
+  /// @dev Emitted when pToken price is changed
+  event PriceUpdated(address owner, uint256 newPrice);
 
-  event Burned(address owner, uint256 amountBurned);
+  /**
+   * @dev Modifier that implements transfer restrictions
+   */
+  modifier restricted(address _from, address _to) {
+    // This contract can send tokens to anyone. If the sender is anyone except this contract, the
+    // only allowed recipient is this contract
+    if (_from != address(this)) {
+      require(_to == address(this), "PToken: Invalid recipient");
```

Comment:
One minor nitpick— why use an if statement here? Why not just:

```solidity
require(_from != address(this) || _to == address(this), "PToken: Invalid recipient");
```

Or, more readably:

```solidity
bool isToOrFromContract = _from != address(this) || _to == address(this);
require(isToOrFromContract, "PToken: Invalid recipient");
```

Reply:
Updated in 3dabd14